fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -295,14 +316,18 @@ byte[] partitionMetadataKeyFromTimestamp(@Nonnull Tuple groupKey, long timestamp
     void savePartitionMetadata(@Nonnull Tuple groupKey, @Nonnull final LucenePartitionInfoProto.LucenePartitionInfo.Builder builder) {
         LucenePartitionInfoProto.LucenePartitionInfo updatedPartition = builder.build();
         state.context.ensureActive().set(
-                partitionMetadataKeyFromTimestamp(groupKey, Tuple.fromBytes(builder.getFrom().toByteArray()).getLong(0)),
+                partitionMetadataKeyFromTimestamp(groupKey, getTimestampFromPartitionInfo(builder)),
                 updatedPartition.toByteArray());
     }
 
+    private static long getTimestampFromPartitionInfo(@Nonnull final LucenePartitionInfoProto.LucenePartitionInfoOrBuilder builder) {
+        return Tuple.fromBytes(builder.getFrom().toByteArray()).getLong(0);
+    }
+
     @Nonnull
     CompletableFuture<LucenePartitionInfoProto.LucenePartitionInfo> findPartitionInfo(@Nonnull Tuple groupKey, long timestamp) {
-        Range range = new Range(state.indexSubspace.subspace(Tuple.from(groupKey, PARTITION_META_SUBSPACE)).pack(),
-                state.indexSubspace.subspace(Tuple.from(groupKey, PARTITION_META_SUBSPACE, timestamp)).pack());
+        Range range = new Range(state.indexSubspace.subspace(groupKey.add(PARTITION_META_SUBSPACE)).pack(),
```

Comment:
In testing I noticed that the keys were having nested tuples. This doesn't seem right to me, so I changed it to use `add` so that it is, e.g. `[g0, 1, 23498]` instead of `[[g0], 1, 23498]`. Since key expressions have a fixed number of columns, the nesting of the tuple was just using extra bytes.